player/src/main/java/com/focusit/jsflight/player/script/PlayerScriptProcessor.java

```diff
@@ -40,123 +39,101 @@ public PlayerScriptProcessor(UserScenario scenario)
         this.scenario = scenario;
     }
 
+    public static Map<String, Object> getEmptyBindingsMap()
+    {
+        return new HashMap<>();
+    }
+
     public boolean executeSelectDeterminerScript(String script, WebDriver wd, WebElement element)
     {
-        Binding binding = getBasicBinding();
-        binding.setVariable("webdriver", wd);
-        binding.setVariable("element", element);
+        Map<String, Object> binding = getEmptyBindingsMap();
+        binding.put(ScriptBindingConstants.WEB_DRIVER, wd);
+        binding.put(ScriptBindingConstants.ELEMENT, element);
 
-        Script scr = engine.getThreadBindedScript(script);
-        scr.setBinding(binding);
-        return (boolean)scr.run();
+        return executeGroovyScript(script, binding, Boolean.class);
     }
 
     /**
-     *
      * @param script
      * @param currentEvent
      * @param prevEvent
      * @return true if currentEvent duplicates prevEvent and should be skipped
      */
     public boolean executeDuplicateHandlerScript(String script, JSONObject currentEvent, JSONObject prevEvent)
     {
-        Binding binding = getBasicBinding();
-        binding.setVariable("current", currentEvent);
-        binding.setVariable("previous", prevEvent);
-        Script scr = engine.getThreadBindedScript(script);
+        Map<String, Object> binding = getEmptyBindingsMap();
+        binding.put(ScriptBindingConstants.CURRENT, currentEvent);
+        binding.put(ScriptBindingConstants.PREVIOUS, prevEvent);
 
-        if (scr == null)
+        try
+        {
+            return executeGroovyScript(script, binding, Boolean.class);
+        }
+        catch (Throwable e)
         {
+            LOG.warn("Failed to create duplicateHandler script. Default value is false");
```

Comment:
It might be much better to log that way 
log.warn("...some text...", e). 

I mean do  not lose an exception when loggin something in a catch block